src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ContigAligner.java

```diff
@@ -0,0 +1,272 @@
+     * Alignment regions:
+     * |---------1:100-200------------|
+     *                                |-----------2:100-200------------------|
+     *                                                                       |----------3:100-200-----------------|
+     * Assmbled breakpoints:
+     * 1) links 1:100-200 to 2:100-200
+     * 2) links 2:100-200 to 3:100-200
+     *
+     * Inserted sequence contains portions of the contig that are aligned to neither region, and therefore may be inserted in
+     * the sample. For example, a translocation breakpoint with a microinsertion:
+     *
+     * Contig:
+     * ACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTG
+     * Alignment regions:
+     * |-----1:100-200-------|
+     *                         |-----2:100-200-----|
+     * Inserted sequence:
+     *  GA
+     *
+     * Homology represents ambiguity about the exact location of the breakpoint. For example, in this case one alignment
+     * region ends with "AC" and the next begins with AC, so we don't know if the AC truly belongs with the first or
+     * second alignment region.
+     *
+     * Contig:
+     * ACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTG
+     * Alignment regions:
+     * |-----1:100-200-------|
+     *                    |-----2:100-200----------|
+     * Homology:
+     *  AC
+     */
+    static class AssembledBreakpoint {
+        String contigId;
+        AlignmentRegion region1;
+        AlignmentRegion region2;
+        String insertedSequence;
+        String homology;
+
+        @Override
+        public String toString() {
+            final StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder.append(contigId);
+            stringBuilder.append("\t");
+            stringBuilder.append(region1.toString());
+            stringBuilder.append("\t");
+            stringBuilder.append(region2.toString());
+            stringBuilder.append("\t");
+            stringBuilder.append(insertedSequence);
+            stringBuilder.append("\t");
+            stringBuilder.append(homology);
+            return stringBuilder.toString();
+        }
+    }
+
+    static class AlignmentRegion {
+
+        final Cigar cigar;
+        final boolean forwardStrand;
+        final SimpleInterval interval;
+        final int mqual;
+        final int startInContig;
+        final int endInContig;
+        final int contigLength;
+
+
+        public AlignmentRegion(final AlnRgn alnRgn) {
+            this.forwardStrand = alnRgn.getStrand() == '+';
+            this.cigar = TextCigarCodec.decode(alnRgn.getCigar());
+            this.interval = new SimpleInterval(alnRgn.getChrom(), (int) alnRgn.getPos() + 1, (int) (alnRgn.getPos() + 1 + cigar.getReferenceLength()));
+            this.mqual = alnRgn.getMQual();
+            this.contigLength = cigar.getReadLength();
+            this.startInContig = startOfAlignmentInContig();
+            this.endInContig = endOfAlignmentInContig();
```

Comment:
My personal feeling is that calling instance methods from the constructor is too error prone.  You're using partial object state to set the rest of the state, and the order becomes crucially important and easy to overlook when maintaining the code.  I'd make the three private methods static, which means you'll have to pass them a forwardStrand flag and a cigar string.  That way they're independent of the state of the object and will always work as intended.